otx/hpo/utils.py

```diff
@@ -0,0 +1,124 @@
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from typing import Any, List, Optional, Tuple, Union
+
+
+def check_type(
+    value: type,
+    available_type: Union[List, Tuple],
+    variable_name: Optional[str] = None,
+    error_message: Optional[str] = None,
+):
+    """Validate value type. If type is different, raise an error.
+
+    Args:
+        value (type): value to type check.
+        available_type (Union[List, Tuple]): types expected that the value has.
+        variable_name (Optional[str], optional): name of value. It's used for error message. Defaults to None.
+        error_message (Optional[str], optional): Error message to use when type is different. Defaults to None.
+
+    Raises:
+        TypeError: If type of value is different with the available type, then error is raised.
+    """
+    if not isinstance(value, available_type):  # type: ignore
+        if not isinstance(available_type, tuple):
+            available_type = [available_type]
+        if error_message is not None:
+            message = error_message
+        elif variable_name is not None:
+            message = (
+                f"{variable_name} should be "
+                + " or ".join([str(val) for val in available_type])
+                + f". Current {variable_name} type is {type(value)}"
+            )
+        else:
+            raise TypeError
+        raise TypeError(message)
+
+
+def check_positive(value: Any, variable_name: Optional[str] = None, error_message: Optional[str] = None):
+    """Validate that value is positivle.
+
+    Args:
+        value (Any): value to validate.
+        variable_name (Optional[str], optional): name of value. It's used for error message. Defaults to None.
+        error_message (Optional[str], optional): Error message to use when type is different. Defaults to None.
+
+    Raises:
+        ValueError: If value isn't positive, the error is raised.
+    """
+    if value <= 0:
+        if error_message is not None:
+            message = error_message
+        elif variable_name:
+            message = f"{variable_name} should be positive.\n" f"your value : {value}"
+        else:
+            raise ValueError
+        raise ValueError(message)
+
+
+def check_not_negative(value: Any, variable_name: Optional[str] = None, error_message: Optional[str] = None):
+    """Validate that value isn't negative.
+
+    Args:
+        value (Any): value to validate.
+        variable_name (Optional[str], optional): name of value. It's used for error message. Defaults to None.
+        error_message (Optional[str], optional): Error message to use when type is different. Defaults to None.
+
+    Raises:
+        ValueError: If value is negative, the error is raised.
+    """
+    if value < 0:
```

Comment:
Hmm.. can we merge `check_positive` and `check_not_negative` in one function? or.. we can design the function that checks the variable is zero or not.

Reply:
I think there is no reason to merge it, because as far as I know, the smaller a function, the better. And I also think current implementation is better to understand than merging it.